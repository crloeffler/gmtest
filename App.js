/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { useState } from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
  ActivityIndicator,
  TouchableOpacity
} from 'react-native';

import {
  Colors,
} from 'react-native/Libraries/NewAppScreen';


export const SongRow = ({ artistName, trackName, collectionPrice, primaryGenreName }) => {
  return (
    <View style={styles.result}>
      <View style={styles.resultRow}>
        <Text>Artist </Text>
        <Text testId="artistName">{artistName}</Text>
      </View>

      <View style={styles.resultRow}>
        <Text>trackName </Text>
        <Text testId="trackName">{trackName}</Text>
      </View>
      <View style={styles.resultRow}>
        <Text>TrackPrice </Text>
        <Text testId="collectionPrice">{collectionPrice}</Text>
      </View>
      <View style={styles.resultRow}>
        <Text>primaryGenreName </Text>
        <Text testId="primaryGenreName">{primaryGenreName}</Text>
      </View>

    </View>
  )
}

const App = () => {
  const [data, setData] = useState({});
  const [loading, setLoading] = useState(false);

  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };
  const getInfo = () => {
    setData({});
    setLoading(true)
    fetch('https://itunes.apple.com/search?term=prince')
      .then(response => response.json())
      .then(data => {
        console.log(data);
        setData(data);
        setLoading(false)
      });
  }
  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <TouchableOpacity onPress={getInfo}>
          <Text testId="show_records">Show Records</Text>
        </TouchableOpacity>

        {loading && <ActivityIndicator size="large" />}
        {data.results?.map((song) =>
          <SongRow artistName={song.artistName} trackName={song.trackName} collectionPrice={song.collectionPrice} primaryGenreName={song.primaryGenreName} />
        )}
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  result: {
    margin: 20,
  },
  resultRow: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  }
});

export default App;
