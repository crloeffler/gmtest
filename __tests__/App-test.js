/**
 * @format
 */

import 'react-native';
import React from 'react';
import App, { SongRow } from '../App';


import renderer from 'react-test-renderer';
const tree = renderer.create(<App />)

const testIds = {
  artistName: 'artistName',
  trackName: 'trackName',
  collectionPrice: 'collectionPrice',
  primaryGenreName: 'primaryGenreName',
}


it('snapshot', () => {
  expect(tree).toMatchSnapshot();
});


test('pass args to song row', () => {

  const artistName = 'prince';
  const trackName = 'purpleRain';
  const collectionPrice = 1.00;
  const primaryGenreName = 'r&b';

  const testRenderer = renderer.create(<SongRow artistName={artistName} trackName={trackName} collectionPrice={collectionPrice} primaryGenreName={primaryGenreName} />);
  const testInstance = testRenderer.root;

  const artistField = testInstance.findByProps({ testId: testIds.artistName }).props
  const trackField = testInstance.findByProps({ testId: testIds.trackName }).props
  const priceField = testInstance.findByProps({ testId: testIds.collectionPrice }).props
  const genereField = testInstance.findByProps({ testId: testIds.primaryGenreName }).props

  expect(artistField.children).toEqual(artistName);
  expect(trackField.children).toEqual(trackName);
  expect(priceField.children).toEqual(collectionPrice);
  expect(genereField.children).toEqual(primaryGenreName);
});