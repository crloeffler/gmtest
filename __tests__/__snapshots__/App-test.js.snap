// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          testId="show_records"
        >
          Show Records
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
